mohawk/tests.py

```diff
@@ -634,12 +634,112 @@ def test_receive_wrong_content(self):
         wrong_sender = self.sender
         self.receive(content='TAMPERED WITH', sender=wrong_sender)
 
+    def test_expected_unhashed_empty_content(self):
+        # This test sets up a scenario where the receiver will receive empty
+        # strings for content and content_type and no content hash in the auth
+        # header.
+        # This is to account for callers that might provide empty strings for
+        # the payload when in fact there is literally no content. In this case,
+        # mohawk depends on the presence of the content hash in the auth header
+        # to determine how to treat the empty strings: no hash in the header
+        # implies that no hashing is expected to occur on the server.
+        self.receive(content='',
+                     content_type='',
+                     sender_kw=dict(content=EmptyValue,
+                                    content_type=EmptyValue,
+                                    always_hash_content=False))
+
     @raises(MisComputedContentHash)
-    def test_unexpected_unhashed_content(self):
-        self.receive(sender_kw=dict(content=EmptyValue,
+    def test_expected_unhashed_empty_content_with_content_type(self):
```

Comment:
Thanks for adding this case. I'm not totally sure it's the right thing to do (the content is still empty even though there is a content type) but I'm fine with shipping like this. Someone will complain if it poses a problem.